### PR TITLE
[stable/redis-ha] add rbac for redis-ha chart

### LINKED ISSUE #3305: [redis-ha] Add support for RBAC
**Is this a request for help?**:

---

BUG REPORT:

**Version of Helm and Kubernetes**:
Kubernetes v1.8.6
Helm v.v2.7.2

**Which chart**:
redis-ha

**What happened**:
After installing the redis-ha chart with Helm (with all default values) on a cluster with enabled Authorization (RBAC), the reedis server is not available. The logs read:
```
Starting redis launcher       
Setting labels               
Selecting proper service to execute           
podIP 100.96.3.4                         
Looking for pods running as master
Error from server (Forbidden): pods is forbidden: User "system:serviceaccount:default:default" cannot list pods in the namespace "default"
Error from server (Forbidden): pods "redisha-redis-ha-server-75c6b89cdc-bzmxm" is forbidden: User "system:serviceaccount:default:default" cannot get pods in the namespace "default"
No masters found: "" Electing first master...
Could not connect to Redis at 127.0.0.1:6379: Connection refused                        
Error from server (Forbidden): pods is forbidden: User "system:serviceaccount:default:default" cannot list pods in the namespace "default"
Electing  master 
Error from server (Forbidden): pods "redisha-redis-ha-server-75c6b89cdc-bzmxm" is forbidden: User "system:serviceaccount:default:default" cannot get pods in the namespace "default"

```

**What you expected to happen**:
The redis service to be available.

**How to reproduce it** (as minimally and precisely as possible):
Against an RBAC-enabled (https://kubernetes.io/docs/admin/authorization/rbac/) cluster
`helm install redis-ha`

### PR DESCRIPTION
**Pull request by smelchior**
The PR adds and option to create the required RBAC elements. This should fix #3305

### CODE REVIEW

File: stable/redis-ha/README.md [hunk 1]
```diff
@@ -70,6 +70,7 @@ The following tables lists the configurable parameters of the Redis chart and th
 | `tolerations`                    | Toleration labels for pod assignment                  | []                                                        |
 | `servers.serviceType`            | Set to "LoadBalancer" to enable access from the VPC   | ClusterIP                                                 |
 | `servers.annotations`            | See Appliance mode                                    | ``                                                        |
+| `rbac.create`                    | create rbac resources                                 | false                                                     |
```
Comment (RoelVanNyen) [inline stable/redis-ha/README.md]: I would personally call this rbac.enabled instead.
If rbac.enabled {
...
}

Comment (smelchior) [reply stable/redis-ha/README.md]: hm, i checked the other charts in the repo and most of them seem to use the rbac.create flag. So for consistency i would prefer to keep the rbac.create.

Comment (gabel) [reply stable/redis-ha/README.md]: @RoelVanNyen looks like @smelchior is right on this -> https://github.com/kubernetes/charts/search?q=%22rbac.create%22&type=Code&utf8=%E2%9C%93

Can this PR go ahead?

**Review by unguiculus — CHANGES_REQUESTED** [review]
Thanks for the PR. Please check the current RBAC best practices and apply them: 
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

**Comment by smelchior** [comment]
Thanks for the hint, i did not know about these policies. They have been applied now.

File: stable/redis-ha/templates/redis-sentinel-deployment.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ spec:
         name: {{ template "fullname" . }}-sentinel
 {{ include "labels.standard" . | indent 8 }}
     spec:
+      {{- if .Values.rbac.create }}
+      serviceAccountName: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/redis-ha/templates/redis-sentinel-deployment.yaml]: `{{ template "redis-ha.serviceAccountName" . }}` and remove the `if` statement around it.

File: stable/redis-ha/templates/redis-server-deployment.yaml [hunk 3]
```diff
@@ -19,6 +19,9 @@ spec:
         redis-node: "true"
 {{ include "labels.standard" . | indent 8 }}
     spec:
+      {{- if .Values.rbac.create }}
+      serviceAccountName: {{ template "fullname" . }}
```
Comment (unguiculus) [inline stable/redis-ha/templates/redis-server-deployment.yaml]: See above

File: stable/redis-ha/README.md (hunk 1 above)
Comment (unguiculus) [inline stable/redis-ha/README.md]: Update the readme

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Review by unguiculus — APPROVED** [review]

**Pull request merged**